server/src/main/java/com/cloud/vm/UserVmManager.java

```diff
@@ -59,6 +59,12 @@ public interface UserVmManager extends UserVmService {
             "Destroys the VM's root volume when the VM is destroyed.",
             true, ConfigKey.Scope.Domain);
 
+    ConfigKey<String> VmDistinctHostNameScope = new ConfigKey<>(String.class, "vm.distinct.hostname.scope", ConfigKey.CATEGORY_ADVANCED,
+            "network",
+            "Scope of resources to check while checking if the hostname is unique. Possible values are global, domain, subdomain, account, network.",
+            true, ConfigKey.Scope.Global, null, "VM distinct hostname scope", null, null, null, ConfigKey.Kind.Select,
+            "global,domain,subdomain,account,network");
+
```

Comment:
Could you please help me understand what's the purpose of this global setting, if VM names have to be unique within a network domain?

Reply:
So, the network domain can be same for networks in different domain, account, etc. If a user a launches a VM with the same name in these networks, there can be conflict. This global setting allows the user to set scope for what all networks we need to check when creating a vm.

Reply:
For project, we specify account_id as the project id. So, this is covered by the account scope. Adding project will make the check complicated.
For networks in a VPC, the network domain is same for all.